src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs

```diff
@@ -751,34 +726,24 @@ async ValueTask CompleteWriteAsync(ValueTask writeTask, byte[] bufferToReturn)
             Dispose(disposing: false);
         }
 
-        // We will only free the read buffer if it
-        // actually contains no decrypted or encrypted bytes
         private void ReturnReadBufferIfEmpty()
         {
-            if (_internalBuffer is byte[] internalBuffer && _decryptedBytesCount == 0 && _internalBufferCount == 0)
+            if (_buffer.ActiveLength == 0)
             {
-                _internalBuffer = null;
-                _internalOffset = 0;
-                _decryptedBytesOffset = 0;
-                ArrayPool<byte>.Shared.Return(internalBuffer);
-            }
-            else if (_decryptedBytesCount == 0)
```

Comment:
I don't know if this matters but the else would put next decrypted data to the beginning of the buffer. 
The encrypted should be always same or bigger so it may not matter.

Reply:
I think it does not matter, since `ResetReadBuffer` is called before receiving and that function moves the data if there is any.